lib/render/RayCaster.cpp

```diff
@@ -407,12 +455,118 @@ RayCaster::UserCoordinates::UpdateFaceAndData( const RayCasterParams* params,
         }
     }
 
+    dataFieldUpToDate = true;
+
     return 0;
 }
 
-void RayCaster::UserCoordinates::FillCoordsXYPlane( const  StructuredGrid* grid,
-                                                    size_t planeIdx,
-                                                    float* coords )
+int  
+RayCaster::UserCoordinates::Update2ndVariable( const RayCasterParams* params,
```

Comment:
There is a lot of code here cut-n-paste from RayCaster::UserCoordinates::UpdateFaceAndData. Please refactor to use helper functions and eliminate the code/functionality duplication.

Reply:
Addressed by commit 625e725